search/first_occurance.py

```diff
@@ -7,16 +7,19 @@
 import unittest
 
 def firstOccurance(array, query):
-    lo, hi = 0, len(array)
-    while lo < hi:
-        mid = lo + (hi - lo) // 2
+    lo, hi = 0, len(array) - 1
+    while lo <= hi:
+        mid = (hi + lo) // 2
```

Comment:
this can cause overflow.
the previous version was right.

Reply:
Restored the previous one.

Reply:
@keon I think there is no Integer overflow in Python. Or is there?

Reply:
@goswami-rahul you're right.
but still good to know for other langs tho.